src/PostgREST/Query/SqlFragment.hs

```diff
@@ -241,10 +241,13 @@ pgFmtCallUnary :: Text -> SQL.Snippet -> SQL.Snippet
 pgFmtCallUnary f x = SQL.sql (encodeUtf8 f) <> "(" <> x <> ")"
 
 pgFmtField :: QualifiedIdentifier -> CoercibleField -> SQL.Snippet
-pgFmtField table CoercibleField{cfName=fn, cfJsonPath=[]} = pgFmtColumn table fn
--- Using to_jsonb instead of to_json to avoid missing operator errors when filtering:
--- "operator does not exist: json = unknown"
-pgFmtField table CoercibleField{cfName=fn, cfJsonPath=jp} = "to_jsonb(" <> pgFmtColumn table fn <> ")" <> pgFmtJsonPath jp
+pgFmtField table CoercibleField{cfName=fn, cfJsonPath=[]}                = pgFmtColumn table fn
+pgFmtField table CoercibleField{cfName=fn, cfIRType=typ, cfJsonPath=jp}  =
+  case typ of
+    "json"  -> pgFmtColumn table fn <> pgFmtJsonPath jp
+    "jsonb" -> pgFmtColumn table fn <> pgFmtJsonPath jp
+    -- Using to_jsonb instead of to_json to avoid missing operator errors when filtering: "operator does not exist: json = unknown"
+    _       -> "to_jsonb(" <> pgFmtColumn table fn <> ")" <> pgFmtJsonPath jp
```

Comment:
This wasn't right. There's another logic that converts the `cfIRType` to `json` here: 

https://github.com/PostgREST/postgrest/blob/e332f038efb46f55d0f463d5ada2f6a812b6410d/src/PostgREST/Plan.hs#L229-L235

It's a bit confusing to have the logic split in two places. Will try to refactor this.

Reply:
Fixing applying the index for order is also another process. Currently not working.